mesonbuild/interpreter.py

```diff
@@ -3251,11 +3292,34 @@ def check_subproject_version(wanted, found):
     def notfound_dependency(self):
         return DependencyHolder(NotFoundDependency(self.environment), self.subproject)
 
-    def get_subproject_dep(self, display_name, dirname, varname, kwargs):
+    def verify_fallback_consistency(self, dirname, varname, cached_dep):
+        subi = self.subprojects.get(dirname)
+        if not cached_dep or not varname or not subi or not cached_dep.found():
+            return
+        dep = subi.get_variable_method([varname], {})
+        if dep.held_object != cached_dep:
+            raise DependencyException('Fallback variable does not match cached dependency')
```

Comment:
This is a bit unclear. Maybe something like "Inconsistency: the dependency has already been overridden, and the fallback variable would define a different dependency object."

Reply:
This happens when you do `dependency('foo', fallback : ['foo', 'foo_dep'])` and the subproject does `meson.override_dependency('foo', another_dep)`. I updated the message, I hope it's a bit better.